src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -266,6 +266,28 @@ void visit_shape_path(Node* node, std::unordered_set<ov::Node*>& visited, std::f
     }
 }
 
+void visit_constant_path(ov::Node* node, std::unordered_set<ov::Node*>& visited, std::function<void(ov::Node*)> func) {
```

Comment:
The function body looks very similar to `visit_shape_path`, maybe is possible to add some internal helper which can implement both functions?
The difference is in continue or assert which could passed as some parameter (predicate)?

Reply:
Good idea. The internal helper reused in both functions